e2e-polybft/e2e/bridge_test.go

```diff
@@ -1444,14 +1444,15 @@ func TestE2E_Bridge_NonMintableERC20Token_WithPremine(t *testing.T) {
 		currentBlock, err := childEthEndpoint.GetBlockByNumber(jsonrpc.LatestBlockNumber, false)
 		require.NoError(t, err)
 
-		// wait for a couple of sprints
-		finalBlockNum := currentBlock.Header.Number + 5*sprintSize
+		// wait for couple of epoches
+		finalBlockNum := currentBlock.Header.Number + uint64(2*epochSize)
+		require.NoError(t, cluster.WaitForBlock(finalBlockNum, 2*time.Minute))
 
 		// the transaction is processed and there should be a success event
 		var stateSyncedResult contractsapi.StateSyncResultEvent
 
-		for i := uint64(0); i < numberOfAttempts; i++ {
-			logs, err := getFilteredLogs(stateSyncedResult.Sig(), 0, finalBlockNum+i*sprintSize, childEthEndpoint)
+		for i := 0; i < numberOfAttempts; i++ {
```

Comment:
I'd still prefer having i as uint64, because that way we wouldn't need to cast it within the for loop two times. Also I'd consider to cast epochSize to `uint64` if feasible.

Reply:
Ok, epochSize is defined as uint64 and casted to int when setting epoch size. Other casts are removed.